gammapy/image/maps.py

```diff
@@ -274,10 +274,24 @@ def coordinates(self, coord_type='skycoord', origin=0, mode='center'):
     def solid_angle(self):
         """
         Solid angle image
+        TODO: description
         """
-        xsky, ysky = self.coordinates(coord_type='galactic', mode='edges')
-        omega = np.abs(np.diff(xsky, axis=1)[1:, :] * np.diff(ysky, axis=0)[:, 1:])
-        return omega.to('sr')
+        coordinates = self.coordinates()
+        lon = coordinates.data.lon
+        lat = coordinates.data.lat
+
+        # Convert to radians
+        lon = np.radians(lon)
```

Comment:
Instead of
```python
lon = coordinates.data.lon
lon = np.radians(lon)
```
you should be able to do this:
```
lon = coordinates.data.lon.radian
```